app/src/org/commcare/activities/connect/ConnectIDNetworkHelper.java

```diff
@@ -192,43 +202,48 @@ private boolean getInternal(Context context, String url, AuthInfo authInfo, Mult
                         ArrayListMultimap.create(),
                         new HashMap<>(),
                         authInfo);
-        getTask.connect(getResponseProcessor(handler));
+        getTask.connect(getResponseProcessor(context, handler));
         getTask.executeParallel();
 
         return true;
     }
 
-    private ConnectorWithHttpResponseProcessor<HttpResponseProcessor> getResponseProcessor(INetworkResultHandler handler) {
+    private ConnectorWithHttpResponseProcessor<HttpResponseProcessor> getResponseProcessor(Context context, INetworkResultHandler handler) {
         return new ConnectorWithHttpResponseProcessor<>() {
             @Override
             public void processSuccess(int responseCode, InputStream responseData) {
                 isBusy = false;
+                dismissProgressDialog(context);
                 handler.processSuccess(responseCode, responseData);
             }
 
             @Override
             public void processClientError(int responseCode) {
                 isBusy = false;
+                dismissProgressDialog(context);
```

Comment:
can abstract this into another internal method something like - `onFinishProcessing()`

Reply:
Done

Reply:
Done